packages/app/src/providers/h5grove/h5grove-api.ts

```diff
@@ -136,91 +166,43 @@ export class H5GroveApi extends DataProviderApi {
   }
 
   public override async getSearchablePaths(path: string): Promise<string[]> {
-    const buffer = await this.fetcher(`/paths/`, { path });
-    return toJSON(buffer) as H5GrovePathsResponse;
-  }
-
-  private async fetchEntity(path: string): Promise<H5GroveEntityResponse> {
     try {
-      const buffer = await this.fetcher(`/meta/`, { path });
-      return toJSON(buffer) as H5GroveEntityResponse;
+      const buffer = await this.fetcher(`${this.baseURL}/paths/`, { path });
+      return toJSON(buffer) as H5GrovePathsResponse;
     } catch (error) {
-      if (!(error instanceof FetcherError)) {
-        throw error;
-      }
-
-      const payload = toJSON(error.buffer);
-      if (!isH5GroveErrorResponse(payload)) {
-        throw error;
-      }
-
-      const { message } = payload;
-      if (message.includes('File not found')) {
-        throw new Error(`File not found: '${this.filepath}'`, { cause: error });
-      }
-      if (message.includes('Permission denied')) {
-        throw new Error(
-          `Cannot read file '${this.filepath}': Permission denied`,
-          { cause: error },
-        );
-      }
-      if (message.includes('not a valid path')) {
-        throw new Error(`No entity found at ${path}`, { cause: error });
-      }
-      if (message.includes('Cannot resolve')) {
-        throw new Error(`Could not resolve soft link at ${path}`, {
-          cause: error,
-        });
-      }
-
-      throw error;
+      throw this.getH5GroveError(error, path) || error;
     }
   }
 
-  private async fetchAttrValues(
-    path: string,
-  ): Promise<H5GroveAttrValuesResponse> {
-    const buffer = await this.fetcher(`/attr/`, { path });
-    return toJSON(buffer) as H5GroveAttrValuesResponse;
-  }
+  private getH5GroveError(error: unknown, path: string): Error | undefined {
```

Comment:
I would perhaps rename this function with another verb than `get` to distinguish it from the other `get..` functions that do fetchs